### PR TITLE
Adjust hypershift-agent-create-proxy

### PR DESCRIPTION
**Pull request by LiangquanLi930**
Adjust step:hypershift-agent-create-proxy to support ipv4, ipv6 and  domain

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

File: ci-operator/step-registry/hypershift/agent/create/proxy/hypershift-agent-create-proxy-commands.sh [hunk 1]
```diff
@@ -9,8 +9,20 @@ source "${SHARED_DIR}/packet-conf.sh" && scp "${SSHOPTS[@]}" "${SHARED_DIR}/nest
...
+  echo "It is an IPv4 address: \${API_SERVER}"
+  EXTRACTED_API_SERVER=$(echo "\${API_SERVER}" | grep -oP '[0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+')
+elif [[ "\${API_SERVER}" =~ ^[0-9a-fA-F:]+\$ ]]; then
+  echo "It is an IPv6 address: \${API_SERVER}"
+  EXTRACTED_API_SERVER=$(echo "\${API_SERVER}" | grep -oP '(?<=://\\[)[^]]+')
+else
+  echo "It is a domain address: \${API_SERVER}"
+  EXTRACTED_API_SERVER=$(echo "\${API_SERVER}" | grep -oP '(?<=://)([^:/]+)')
+fi
+sed -i "1 s|\$| \$EXTRACTED_API_SERVER|" \$HOME/squid.conf
```
Comment (aleskandro) [inline ci-operator/step-registry/hypershift/agent/create/proxy/hypershift-agent-create-proxy-commands.sh:25]:
```suggestion
ssh "${SSHOPTS[@]}" "root@${IP}" bash - << 'EOF' |& sed -e 's/.*auths.*/*** PULL_SECRET ***/g'
set -x

API_SERVER=$(cat nested_kubeconfig | yq ".clusters[0].cluster.server")
EXTRACTED_API_SERVER=""
if [[ "${API_SERVER}" =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
  echo "It is an IPv4 address: ${API_SERVER}"
  EXTRACTED_API_SERVER=$(echo "${API_SERVER}" | grep -oP '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+')
elif [[ "${API_SERVER}" =~ ^[0-9a-fA-F:]+$ ]]; then
  echo "It is an IPv6 address: ${API_SERVER}"
  EXTRACTED_API_SERVER=$(echo "${API_SERVER}" | grep -oP '(?<=://\[)[^]]+')
else
  echo "It is a domain address: ${API_SERVER}"
  EXTRACTED_API_SERVER=$(echo "${API_SERVER}" | grep -oP '(?<=://)([^:/]+)')
fi
sed -i "1 s|$| $EXTRACTED_API_SERVER|" $HOME/squid.conf
```

And

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-conformance

**Review by jparrill — COMMENTED** [review]
/lgtm

**Pull request merged**